server/s3_object_metadata.h

```diff
@@ -268,7 +268,9 @@ class S3ObjectMetadata : private S3ObjectMetadataCopyable {
   const struct s3_motr_idx_layout& get_part_index_layout() const {
     return part_index_layout;
   }
-
+  // This is temporary change, will be removed once
+  // delete object changes related delete marker are merged
+  virtual bool get_delete_marker() { return false; }
   void regenerate_version_id();
 
   std::string const get_obj_version_id() { return object_version_id; }
```

Comment:
can't be a const reference

Reply:
If the comment is about `get_obj_version_id()`, there's no reason to change it in this PR, as it's completely unrelated. FYI, I have already fixed this in the version ID change I've been working on.

However, `get_delete_marker()` should be const.

```c++
virtual bool get_delete_marker() const { return false; }
```

Reply:
Code is rebased to latest object-versioning branch which has this api implemented. Updated code accordingly.